src/js/actions/libraries.js

```diff
@@ -499,6 +497,145 @@ define(function (require, exports) {
     };
     renameAsset.reads = [];
     renameAsset.writes = [locks.CC_LIBRARIES, locks.JS_LIBRARIES];
+    
+    /**
+     * Open the specified graphic asset for edit by creating a temp copy of the asset's document and open
+     * in Photoshop. The library store will listen to the document save and close event. 
+     * 
+     * @param {AdobeLibraryElement} element
+     * @return {Promise}
+     */
+    var openGraphicForEdit = function (element) {
```

Comment:
You also need to transfer to this from `actions/superselect.js#editLayer`, when it's a library linked smart object. Double clicking a such layer will work that way.